### DISCUSSION TITLE
Issue with asset/modules, while loading the image through css, with frontend and backend server being different.

### DISCUSSION THREAD

**Discussion by Chinmay2709** [webpack/webpack#18437] [Q&A] [answered]
Before I explain the issue that I have in my case, here is my general info:
![webpack-cli INFO](https://github.com/webpack/webpack/assets/107140611/3dc47c48-4d31-417c-b6f0-f12f76d30afc)

**Now, here is my project description:**

- I am developing a Vue application.
- I am using tailwind-css in my application.
- my back-end is on python flask.
- I am using xampp apache to serve my application on : ( URL:  http://127.0.0.1:80/)
- And my Front-end web-pack server runs on: (URL: http://127.0.0.1:8080/bundle.js)
- Therefore my Project structure is something like this:

![project-struc](https://github.com/webpack/webpack/assets/107140611/5aa63dae-4ab6-4421-812a-a7cc3ae99033)

**Now coming back to problem.**
So far everything was working fine, since I'm using web-pack V-5, I was aware that the url-loader and file-loader are depreciated, and instead asset/modules are introduced. Therefore, that's what I used, here is my webpack.config.js :

![webpackconfig](https://github.com/webpack/webpack/assets/107140611/660f8187-c254-436c-a41e-a47578ae6734)

Even after that when I run my project I, couldn't load my Image to the project:

![Browser debugger image prob](https://github.com/webpack/webpack/assets/107140611/a744987a-0042-4af9-a60c-6cf6d721de6b)

Now, here is what I found, the problem was never with web-pack or with my path inside my main.css file which has property(background-image) with url for the image. I found that the problem is with the dynamic URL with hash for image that is loaded by the web-pack, since I am using a different PORT ( 8080 ) for my bundle.js, the web-pack loads the file with URL without the port number, which creates a problem. Here, is how when I change or add port number to URL, image seems to be loading.

![Screenshot 2024-05-21 183755](https://github.com/webpack/webpack/assets/107140611/9b3e44d4-a4d1-4b54-9ed9-6a43e71e65a7)

**Now here is my question:**

Is there a way I can give correct port to the web-pack so that it knows the correct URL to reach my asset ? or if not then what should be my approach, given that my back-end would remain on python, since I have specific use for the language. 

Thanks~

**Comment by alexander-akait** [comment]
What is `publicPath`? Can you provide webpack.config.js file?

**Reply by Chinmay2709** [reply]
# This is my webpack.config.js and my " publicPath=' /public/ ' "

`
const path = require('path');
const { VueLoaderPlugin, default: loader } = require('vue-loader');
const webpack = require('webpack');

module.exports = {

    entry: '/src/main.js',
    devServer: {
        port :8080
    },
    output: {

        path: path.join(__dirname,'/dist'),
        filename: 'bundle.js',
        publicPath: "/public/",
        assetModuleFilename: 'images/[hash][ext][query]'

    },
    module: {
        
        rules: [
            {
                test: /\.vue$/,
                exclude: /node_modules/,
                loader: 'vue-loader'
            },

            {
                test: /\.css$/,
                use: [
                    
                    'style-loader',
                    'css-loader',
                    'postcss-loader',

                ],
            },

            {
                test: /\.(png)$/,
                type: 'asset/resource'
            }
        ]

    },

    plugins: [

        new VueLoaderPlugin(),
        new webpack.ProvidePlugin({
            
            $: 'jquery',
            jQuery: 'jquery'

        })

    ],

};
`

**Reply by alexander-akait** [reply, marked as answer]
You can just remove `publicPath` and so it will be auto and always relative

**Reply by Chinmay2709** [reply]
Thanks Alexander. Sorry to revert back to you so late. It worked fine for me. 🫰🏻
